sdk/storage/Azure.Storage.Common/src/UriExtensions.cs

```diff
@@ -59,5 +59,54 @@ public static Uri AppendToPath(this Uri uri, string segment)
             }
             return parameters;
         }
+
+        /// <summary>
+        /// Get the account name from the domain portion of a Uri.
+        /// </summary>
+        /// <param name="uri">The Uri.</param>
+        /// <param name="serviceSubDomains">Required. The service subdomains used to validate that the
+        /// domain is in the expected format. This should be "blob" for blobs, "file" for files,
+        /// "queue" for queues, "blob" and "dfs" for datalake.</param>
+        /// <returns>Account name</returns>
+        public static string GetAccountNameFromDomain(this Uri uri, params string[] serviceSubDomains)
```

Comment:
Can we make this return `null` if not found?  Then we don't have to take the `params` array, we'll be able to tell if it worked, and we can chain things for DataLake via `uri.GetAccountName("blob") ?? uri.GetAccountName("dfs") ?? ""`.

Reply:
I considered this but figured it would be slightly less performant since we would be doing the account name parsing twice. I agree what you propose is a bit cleaner, but I figured since this is an internal method it probably wasn't worth any performance hit. Having said that, I am fine with changing it if you think that is better.

Reply:
I think doing a couple extra IndexOfs just for DataLake is better than making all the other clients allocate and enumerate when they don't need it.